Plutarch/Builtin.hs

```diff
@@ -170,12 +172,12 @@ instance PListLike PBuiltinList where
 instance (PLift a, PEq a) => PEq (PBuiltinList a) where
   (#==) xs ys = plistEquals # xs # ys
 
-data PData s
-  = PDataConstr (Term s (PBuiltinPair PInteger (PBuiltinList PData)))
-  | PDataMap (Term s (PBuiltinList (PBuiltinPair PData PData)))
-  | PDataList (Term s (PBuiltinList PData))
-  | PDataInteger (Term s PInteger)
-  | PDataByteString (Term s PByteString)
+data PData (s :: S) = PData (Term s PData)
```

Comment:
you could've kept this the way it was and implemented `PlutusType` with `ChooseData` or something. But this is fine too, if slightly confusing to people expecting isomorphism with the `Data` declaration.

Reply:
`PlutusType` wasn't implementable in an efficient  way due to the difference between `UnConstrData` and `ConstrData`. We could possibly do it with but that's Plutarch 2.0.